sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -769,9 +771,9 @@ class SessionCatalog(
         desc = metadata,
         output = metadata.schema.toAttributes,
         child = parser.parsePlan(viewText))
-      SubqueryAlias(table, db, child)
+      SubqueryAlias(multiParts, child)
     } else {
-      SubqueryAlias(table, db, UnresolvedCatalogRelation(metadata))
+      SubqueryAlias(multiParts, UnresolvedCatalogRelation(metadata))
```

Comment:
@cloud-fan Now that session catalog name is inserted as a qualifier, `matchWithTwoOrLessQualifierParts` will not be used for resolving columns for a table. If we need this in Spark 3.0, I could try to update that function to handle 3 parts qualifier `catalog_name.db.table`.

Reply:
Yea let's handle 3 parts qualifier, to avoid perf regression.